ocs_ci/ocs/resources/mcg.py

```diff
@@ -799,20 +799,41 @@ def _check_mirroring():
                     else:
                         results.append(False)
 
-            return all(results)
-
-        try:
-            for mirroring_is_complete in TimeoutSampler(timeout, 5, _check_mirroring):
-                if mirroring_is_complete:
+            return results
+
+        is_mirroring_complete = _check_mirroring()
+        current_percentage = (
+            is_mirroring_complete.count(True) / len(is_mirroring_complete)
+        ) * 100
+        logger.info(f"{current_percentage}% mirroring is done.")
+        previous_percentage = 0
+        if current_percentage == 100:
+            logger.info("All objects mirrored successfully.")
+            return True
+        else:
+            while current_percentage < 100:
+                previous_percentage = current_percentage
+                for i in range(timeout):
+                    is_mirroring_complete = _check_mirroring()
+                    current_percentage = (
+                        is_mirroring_complete.count(True) / len(is_mirroring_complete)
+                    ) * 100
+                    if previous_percentage == current_percentage:
+                        import time
+
+                        time.sleep(3)
+                        if i + 1 == timeout:
+                            logger.error("The mirroring process is stuck.")
+                            assert False
+                    else:
+                        break
+                is_mirroring_complete = _check_mirroring()
+                current_percentage = (
+                    is_mirroring_complete.count(True) / len(is_mirroring_complete)
+                ) * 100
+                if current_percentage == 100:
```

Comment:
This can be omitted by relying on the check of the while loop as described in a previous comment.

Reply:
Fixed